### DISCUSSION TITLE
Conditional forwardTo()

### DISCUSSION THREAD

**Discussion by VanTanev** [statelyai/xstate#1815] [Q&A]
I have the following design:

1. A `main` machine that spawn 3 actors
2. A `sync-machine` machine that is invoked when the `main` machine enters a specific state
3. `child` actors that use `sendParent({ type: 'SYNC_STATUS', data: ... })` to send messages to `main`, which `main` then routes to `sync-machine` like this:

```typescript
{
  on: {
    SYNC_STATUS: { actions: forwardTo('sync-machine') }
  }
}
```

It turns out, I sometimes want to manually change a state in a `child`, but doing so will invoke that state's actions, which triggers `sendParent('SYNC_STATUS')`, which hits `forwardTo()`, which explodes when the `sync-machine` service is not running with:
```
Error: Unable to send event to child 'sync-machine' from service 'resource-manager'.
```

Is there a way to make the `forwardTo` action conditional on the `sync-machine` service existing?
A normal guard does not work, because `sync-machine` is an `invoke`, not a `spawn`, so it does not exist on the context.

**Comment by Andarist** [comment]
You could use `choose` action

**Reply by davidkpiano** [reply]
Link: https://xstate.js.org/docs/guides/actions.html#choose-action

**Reply by VanTanev** [reply]
I don't think `choose` works in my case, because I don't have a `ref` to `sync-machine` on `context`. I would need to do something like, eg, `Object.keys(state.children).includes('sync-machine')`
